sys/net/link_layer/lorawan/Kconfig

```diff
@@ -0,0 +1,476 @@
+        in LoRa PHY.
+
+if KCONFIG_USEMODULE_LORAWAN
+
+choice
+    bool "LoRaWAN region"
+    default LORAMAC_REGION_EU_868
+    help
+        Configure the LoRaWAN region.
+
+config LORAMAC_REGION_EU_868
+    bool "EU863-870"
+
+config LORAMAC_REGION_IN_865
+    bool "IN865-867"
+
+endchoice
+
+choice
+    bool "Join procedure"
+    default LORAMAC_DEFAULT_JOIN_PROCEDURE_OTAA
+    help
+        Configure the join procedure for end node activation. There are two
+        options, Over The Air Activation (OTAA) results in device sending
+        join request to the network whereas in the case of Activation By
+        Personalization (ABP) the user enters the activation keys manually. OTAA
+        is the preferred and most secure way to connect to a LoRaWAN network.
+        For quick testing ABP is preferred as the device can transmit right
+        away without waiting for the network to provision the keys.
+
+config LORAMAC_DEFAULT_JOIN_PROCEDURE_OTAA
+    bool "Over the air activation (OTAA)"
+
+config LORAMAC_DEFAULT_JOIN_PROCEDURE_ABP
+    bool "Activation by personalization (ABP)"
+
+endchoice
+
+if LORAMAC_DEFAULT_JOIN_PROCEDURE_OTAA
+
+config LORAMAC_DEV_EUI_DEFAULT
+    string "Device EUI"
+    default "0000000000000000"
+    help
+        Configure the default device EUI for join procedure.
+
+config LORAMAC_APP_EUI_DEFAULT
+    string "Application EUI"
+    default "0000000000000000"
+    help
+        Configure the default application EUI for join procedure.
+
+config LORAMAC_APP_KEY_DEFAULT
+    string "Application key"
+    default "00000000000000000000000000000000"
+    help
+        Configure the default application key for join procedure.
+
+endif # LORAMAC_DEFAULT_JOIN_PROCEDURE_OTAA
+
+if LORAMAC_DEFAULT_JOIN_PROCEDURE_ABP
+
+config LORAMAC_APP_SKEY_DEFAULT
+    string "Application session key"
+    default "00000000000000000000000000000000"
+    help
+        Configure the default application session key for ABP.
+
+config LORAMAC_NWK_SKEY_DEFAULT
+    string "Network session key"
+    default "00000000000000000000000000000000"
+    help
+        Configure the default network session key for ABP.
+
+config LORAMAC_DEV_ADDR_DEFAULT
+    string "Device EUI"
+    default "00000000"
+    help
+        Configure the default device EUI for ABP.
+
+config LORAMAC_DEFAULT_NETID
+    int "NET ID"
+    depends on LORAMAC_ADVANCED_OPTIONS
+    default 1
+    help
+        Configure the default device NETID. NETID is used by networks for
+        assigning network-specific addresses to their end-devices
+        (i.e., DevAddr) so that uplink frames sent by those devices even when
+        they are roaming outside their home network can be forwarded to their
+        home network.
+
+endif # LORAMAC_DEFAULT_JOIN_PROCEDURE_ABP
+
+choice
+    bool "Device Class"
+    default LORAMAC_DEFAULT_DEVICE_CLASS_A
+    help
+        Configure the class of device. **IMP** GNRC LoRaWAN only supports
```

Comment:
I think these classes should be modeled depending of the support of the lorawan stack (e.g depend on a feature)

Reply:
Sure why not.